docs/source/resources/graphql-glossary.md

```diff
@@ -0,0 +1,353 @@
+
+```js
+type User @auth {
+  name: String!
+  banned: Boolean @auth!
+}
+```
+
+<h2 id="docstring">Docstring</h2>
+<p>It is used for providing descriptions of types, fields and arguments. It adds metadata to a GraphQL document. Docstrings show up in the documentation panel inside GraphQL playground and GraphiQL.</p>
+
+```js
+"""
+Description for the User
+"""
+type User {
+  """
+  Description for first Name
+  """
+  firstName: String!
+
+  age(
+    """
+    Must be an integer
+    """
+    arg: Int
+  )
+}
+```
+
+<h2 id="document">Document</h2>
+<p>A file or request string that contains one or multiple definitions of a GraphQL type system and can be interpreted by a GraphQL execution engine.</p>
+
+<h2 id="extensions">Extensions</h2>
+<p>Special fields in the Graphql response that allows you to attach extra metadata. [Apollo tracing](https://github.com/apollographql/apollo-server/tree/master/packages/apollo-tracing) is an example of an extension. </p>
+
+
+<h2 id="field">Field</h2>
+<p>A unit of data you are asking for in a Schema, which ends up as a field in your JSON response data.</p>
+
+```js
+type Author {
+  id: Int!
+  firstName: String
+  lastName: String
+}
+```
+
+`id`, `firstName`, and `lastName` are fields in the Author type above.
+
+
+<h2 id="fragment">Fragment</h2>
+<p>A selection set that can be reused in multiple query operations. A [GraphQL fragment](https://www.apollographql.com/docs/react/advanced/fragments.html) is a shared piece of query logic.</p>
+
+```js
+fragment UserData on User {
+  id: ID!
+  firstName: String!
+  lastName: String!
+}
+
+query getUsers {
+  allUsers {
+    ...UserData
+  }
+}
+```
+
+<h2 id="gql-function">gql function</h2>
+<p>A [JavaScript template literal tag](https://github.com/apollographql/graphql-tag) that parses GraphQL queries into an abstract syntax tree (AST).</p>
+
+```js
+const typeDefs = gql`
+  type File {
+    filename: String!
+    mimetype: String!
+    encoding: String!
+  }
+`;
+```
+
+<h2 id="graphql-playground">GraphQL Playground</h2>
+<p>An in-browser IDE for GraphQL development and workflow. Added benefits exist such as theme change, automatic schema reloading, HTTP headers configuration, query history and GraphQL subscription support. In addition, it comes [out-of-the-box in Apollo Server 2](https://www.apollographql.com/docs/apollo-server/features/graphql-playground.html).</p>
+
+<h2 id="graphiql">GraphiQL</h2>
+<p>An in-browser IDE for GraphQL development.</p>
+
+<h2 id="introspection">Introspection</h2>
+<p>A technique to provide detailed information about a GraphQL API's schema. Types and fields used in introspection are prefixed with "__" two underscores.</p>
+
+<h2 id="mutation">Mutation</h2>
+<p>An operation for creating, modifying and destroying data.</p>
+
+```js
+mutation AddTodo($type: String!) {
+  addTodo(type: $type) {
+    id
+    type
+  }
+}
+```
+
+<h2 id="normalization">Normalization</h2>
```

Comment:
Can you please make this definition more specific to Apollo Client?

https://www.apollographql.com/docs/react/advanced/caching.html#normalization